cms/admin/placeholderadmin.py

```diff
@@ -856,107 +764,63 @@ def _paste_placeholder(self, request, plugin, target_language,
             plugins=new_plugins,
             target_language=target_language,
             target_placeholder=target_placeholder,
-            target_order=target_tree_order,
         )
         return new_plugins
 
-    def _move_plugin(self, request, plugin, target_language,
-                     target_placeholder, tree_order, target_parent=None):
-        if not self.has_move_plugin_permission(request, plugin, target_placeholder):
+    def _move_plugin(self, request, plugin, target_position, target_placeholder=None, target_parent=None):
+        language = plugin.language
+        source_placeholder = plugin.placeholder
+
+        if not self.has_move_plugin_permission(request, plugin, source_placeholder):
             message = force_text(_("You have no permission to move this plugin"))
             raise PermissionDenied(message)
 
-        plugin_data = {
-            'language': target_language,
-            'placeholder': target_placeholder,
-        }
-
-        source_language = plugin.language
-        source_placeholder = plugin.placeholder
-        source_tree_order = source_placeholder.get_plugin_tree_order(
-            language=source_language,
-            parent_id=plugin.parent_id,
-        )
+        if target_placeholder and not self.has_move_plugin_permission(request, plugin, source_placeholder):
+            message = force_text(_("You have no permission to move this plugin"))
+            raise PermissionDenied(message)
 
         if target_parent:
             target_parent_id = target_parent.pk
         else:
             target_parent_id = None
 
-        if target_placeholder != source_placeholder:
-            target_tree_order = target_placeholder.get_plugin_tree_order(
-                language=target_language,
-                parent_id=target_parent_id,
-            )
-        else:
-            target_tree_order = source_tree_order
-
         action_token = self._send_pre_placeholder_operation(
             request,
             operation=operations.MOVE_PLUGIN,
             plugin=plugin,
-            source_language=source_language,
+            source_language=plugin.language,
             source_placeholder=source_placeholder,
             source_parent_id=plugin.parent_id,
-            source_order=source_tree_order,
-            target_language=target_language,
-            target_placeholder=target_placeholder,
+            target_language=language,
```

Comment:
Why the discrepancy between `source_language` and this? `language` point to the same thing